src/praisonai-bot/praisonai_bot/bots/discord.py

```
@@ -508,6 +509,39 @@ async def send_message(
+            guilds = list(getattr(self._client, "guilds", []) or [])
+        except Exception as e:  # noqa: BLE001 — enumeration is best-effort
+            logger.debug(f"Discord list_channels: could not read guilds: {e}")
+            return []
+
+        for guild in guilds:
+            guild_name = getattr(guild, "name", "") or ""
+            for ch in getattr(guild, "text_channels", []) or []:
+                try:
+                    ch_id = str(ch.id)
+                except Exception:
+                    continue
```

Ruff (0.15.20)

[error] `try`-`except`-`continue` detected, consider logging the exception

(S112)